### PR TITLE
Fixed oscap-ssh failing to retrieve the result files when executing with --sudo

### PR DESCRIPTION
**Pull request by rmetrich**
Depending on the umask configuration of the target system, "sudo oscap"
may create the result files in temporary directory with 600 permissions,
which makes retrieving the log (as the regular user that ssh'ed to the
system) impossible:

~~~
$ oscap-ssh --sudo user@system 22 xccdf eval ...
[...]
oscap exit code: 0
Copying back requested files...
scp: /tmp/tmp.0kfbPWEy6u/report.html: Permission denied
Failed to copy the HTML report back to local machine!
~~~

Scenario to reproduce the failure: set a default umask in /etc/sudoers:

~~~
Defaults	umask = 0077
~~~

The fix consists in changing the result files' ownership from "root" to
user's back, all while in the single sudo (using two sudo commands
wouldn't be nice since the user may get the password prompt twice,
depending on the sudo's configuration).

### CODE REVIEW

**Comment by evgenyz** [comment]
@openscap-ci test this, please

**Comment by rmetrich** [comment]
An even more simple reproducer is to change the umask in `/etc/bashrc`:
~~~
    if [ $UID -gt 199 ] && [ "`/usr/bin/id -gn`" = "`/usr/bin/id -un`" ]; then
       umask 002
    else
       umask 022
    fi
    umask 027          <<<<<< HERE FOR EXAMPLE
~~~

**Comment by matejak** [comment]
ok to test

File: utils/oscap-ssh [hunk 1]
```diff
@@ -279,7 +279,11 @@ echo "Starting the evaluation..."
 # changing directory because of --oval-results support. oval results files are
 # dumped into PWD, and we can't be sure by the file names - we need controlled
 # environment
-ssh_execute_with_command_and_options "cd $REMOTE_TEMP_DIR; $OSCAP_SUDO oscap $(command_array_to_string oscap_args)" "$SSH_TTY_ALLOCATION_OPTION"
+if [ -z "$OSCAP_SUDO" ]; then
+    ssh_execute_with_command_and_options "cd $REMOTE_TEMP_DIR; oscap $(command_array_to_string oscap_args)" "$SSH_TTY_ALLOCATION_OPTION"
+else
+    ssh_execute_with_command_and_options "cd $REMOTE_TEMP_DIR; $OSCAP_SUDO sh -c 'oscap $(command_array_to_string oscap_args); rc=\$?; chown \$SUDO_USER $REMOTE_TEMP_DIR/*; exit \$rc'" "$SSH_TTY_ALLOCATION_OPTION"
```
Comment (yuumasato) [inline utils/oscap-ssh]: Please, break the line to improve readability.

Comment (rmetrich) [reply utils/oscap-ssh]: @yuumasato Is the proposed split of line ok for you?

Comment (yuumasato) [reply utils/oscap-ssh]: Yes, thank you.

**Comment by yuumasato** [comment]
@rmetrich Sorry, I noticed just now that you proposed the PR against master.
This is a bugfix that can actually go into maint-1.2 branch. Could you propose the PR against `maint-1.2`?
Editing the base branch in "Edit" button above may be sufficient. If it doesn't work, you may need to rebase your commits and make new PR.
Sorry for the hassle.

**Comment by yuumasato** [comment]
Closing in favor of #1485.

**Comment by rmetrich** [comment]
@yuumasato Wondering what is the exact process to merge a PR.
Is it first in maint-1.2, then merged automatically in master?
Or master is not used at all?

**Comment by yuumasato** [comment]
@rmetrich OpenSCAP does forward porting of fixes, instead of back porting.
More info in [versioning doc](https://github.com/OpenSCAP/openscap/blob/11eb193faaf5858e50ed6906e9ca1f6f63449493/docs/contribute/versioning.adoc#all-applied-changes-must-be-propagated)

`maint-1.2` branch will be merge in `maint-1.3`, and then to `master`, so all branches have the bugfix.
Not sure what is the criteria or how often though.
@jan-cerny do we merge the branches whenever there are bugfixes to forward port?

**Comment by evgenyz** [comment]
We usually do this before release.

**Comment by jan-cerny** [comment]
@yuumasato you can merge branches any time you wish, it has no criteria

**Comment by yuumasato** [comment]
Merged https://github.com/OpenSCAP/openscap/commit/8102d7bf1e67608f5d5203c1367aae1e47da5597

**Pull request closed without merging**